core/src/View/Kokkos_BasicView.hpp

```diff
@@ -0,0 +1,260 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_IMPL_PUBLIC_INCLUDE
+static_assert(false,
+              "Including non-public Kokkos header files is not allowed.");
+#endif
+
+#ifndef KOKKOS_BASIC_VIEW_HPP
+#define KOKKOS_BASIC_VIEW_HPP
+#include <type_traits>
+#include <Kokkos_Macros.hpp>
+#include <impl/Kokkos_Utilities.hpp>
+#include <impl/Kokkos_SharedAlloc.hpp>
+#include "Kokkos_ViewAlloc.hpp"
+#include "MDSpan/Kokkos_MDSpan_Accessor.hpp"
+#include "MDSpan/Kokkos_MDSpan_Header.hpp"
+
+namespace Kokkos {
+namespace Impl {
+template <class L>
+struct kokkos_layout_conversion {
+  using type = L;
+};
+
+template <>
+struct kokkos_layout_conversion<LayoutLeft> {
+  using type = Experimental::layout_left_padded<dynamic_extent>;
+};
+
+template <>
+struct kokkos_layout_conversion<LayoutRight> {
+  using type = Experimental::layout_right_padded<dynamic_extent>;
+};
+
+template <typename ExecutionSpace>
+using preferred_layout_for_exec_space = typename ExecutionSpace::array_layout;
+}  // namespace Impl
+
+template <class ElementType, class Extents,
+          class ExecutionSpace = Kokkos::DefaultExecutionSpace,
+          class MemorySpace    = typename ExecutionSpace::memory_space,
+          class LayoutPolicy =
+              Impl::preferred_layout_for_exec_space<ExecutionSpace>>
+class BasicView {
+ private:
+  using internal_layout_type =
+      typename Impl::kokkos_layout_conversion<LayoutPolicy>::type;
+
+  using default_alloc_params = Impl::alloc_params<MemorySpace, ExecutionSpace>;
+
+ public:
+  using extents_type  = Extents;
+  using accessor_type = default_accessor<ElementType>;
```

Comment:
so my feeling is that we should use a `Kokkos::SpaceAwareAccessor<DeviceType, MemoryTraits>` as the standard thing.